### PR TITLE
python3Packages.{confuse,mediafile}: init

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
These will soon be needed for a new release of beets

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/confuse/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{ buildPythonPackage
+, enum34
+, fetchPypi
+, lib
+, pythonOlder
+, pythonPackages
+}:
```
Comment (jonringer) [inline pkgs/development/python-modules/confuse/default.nix:7]:
```suggestion
}:

```

File: pkgs/development/python-modules/confuse/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+buildPythonPackage rec {
+  pname = "confuse";
+  version = "1.3.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "07rbsd9l5widsirbn6rcfadh7gb2bdy2frhlga7bhxdizmhir2pn";
+  };
+
+  propagatedBuildInputs = [ pythonPackages.pyyaml ] ++ lib.optional (pythonOlder "3.4") enum34;
```
Comment (jonringer) [inline pkgs/development/python-modules/confuse/default.nix:17]: there will never be a <3.4 interpreter other than 2.7, and python2.7 is disabled anyway
```suggestion
  propagatedBuildInputs = [ pyyaml ];
```

File: pkgs/development/python-modules/mediafile/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ buildPythonPackage
+, enum34
+, fetchPypi
+, lib
+, pythonOlder
+, pythonPackages
```
Comment (jonringer) [inline pkgs/development/python-modules/mediafile/default.nix:6]: don't import pythonPackages while in `python-modules, just import the individual packages
```suggestion
```

File: pkgs/development/python-modules/mediafile/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "14vqvv4n0d93hbz9xpazcy3k7bfz3lapicp0qamjm656bi77j6fs";
+  };
+
+  propagatedBuildInputs = [
+    pythonPackages.mutagen
+    pythonPackages.six
+    pythonPackages.nose
```
Comment (jonringer) [inline pkgs/development/python-modules/mediafile/default.nix:20]:
```suggestion
    mutagen
    six
    nose
```

File: pkgs/development/python-modules/mediafile/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  ] ++ lib.optional (pythonOlder "3.4") enum34;
```
Comment (jonringer) [inline pkgs/development/python-modules/mediafile/default.nix:21]: you disabled the python27 scenario anyway (no <3.4 interpreter will ever be included)
```suggestion
  ];
```

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -936,6 +938,8 @@ in {
 
   maxminddb = callPackage ../development/python-modules/maxminddb { };
 
+  mediafile = disabledIf (!isPy3k) (callPackage ../development/python-modules/mediafile { });
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:941]: move disable logic into the expression (this is specific to `buildPythonX`)
```suggestion
  mediafile = callPackage ../development/python-modules/mediafile { };
```

File: pkgs/development/python-modules/mediafile/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{ buildPythonPackage
+, enum34
+, fetchPypi
+, lib
+, pythonOlder
+, pythonPackages
+}:
+buildPythonPackage rec {
+  pname = "mediafile";
+  version = "0.5.0";
```
Comment (jonringer) [inline pkgs/development/python-modules/mediafile/default.nix:10]:
```suggestion
  version = "0.5.0";
  disabled = isPy27;
```

File: pkgs/top-level/python-packages.nix [hunk 8]
```diff
@@ -601,6 +601,8 @@ in {
 
   compiledb = callPackage ../development/python-modules/compiledb { };
 
+  confuse = disabledIf (!isPy3k) (callPackage ../development/python-modules/confuse { });
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:604]: move this into expression as well
```suggestion
  confuse = callPackage ../development/python-modules/confuse { };
```

File: pkgs/development/python-modules/confuse/default.nix [hunk 9]
```diff
@@ -0,0 +1,25 @@
+{ buildPythonPackage
+, enum34
+, fetchPypi
+, lib
+, pythonOlder
+, pythonPackages
+}:
+buildPythonPackage rec {
+  pname = "confuse";
+  version = "1.3.0";
```
Comment (jonringer) [inline pkgs/development/python-modules/confuse/default.nix:10]:
```suggestion
  version = "1.3.0";
  disabled = isPy27;
```

File: pkgs/development/python-modules/confuse/default.nix [hunk 10]
```diff
@@ -0,0 +1,25 @@
+{ buildPythonPackage
+, enum34
```
Comment (jonringer) [inline pkgs/development/python-modules/confuse/default.nix:2]:
```suggestion
, isPy27
```

File: pkgs/development/python-modules/confuse/default.nix [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+, fetchPypi
+, lib
+, pythonOlder
+, pythonPackages
```
Comment (jonringer) [inline pkgs/development/python-modules/confuse/default.nix:6]:
```suggestion
, pyyaml
```

**Comment by lovesegfault** [comment]
@jonringer Fastest review in the west! I think I got everything :)

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 91789` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.confuse</li>
    <li>python37Packages.mediafile</li>
    <li>python38Packages.confuse</li>
    <li>python38Packages.mediafile</li>
    <li>python39Packages.confuse</li>
    <li>python39Packages.mediafile</li>
  </ul>
</details>

**Pull request merged**